### PR TITLE
Fix addable types constrain for repository folders for API calls

### LINKED ISSUE #2303: Dossiers in Ordnungspositionen, die nicht Rubriken sind
Offenbar können in der auf lab.onegovgever.ch aufgeschalteten Version der API Dossiers auf Ordnungspositionen erstellt werden, die nicht Rubriken sind:
https://lab.onegovgever.ch/ordnungssystem/gesundheit/dossier-323#overview

Dies sollte nicht möglich sein.

### PR DESCRIPTION
**Pull request by buchi**
Implement an IConstrainTypes adapter to restrict addable types in repository
folders instead of overwriting the allowedContentTypes method in the content
type. This way the constrains are also checked in API calls (e.g. invokeFactory).

Fixes #2303

### CODE REVIEW

File: opengever/repository/constrains.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        # 1. If this RepositoryFolder contains another RF, we should not be
+        # able to add other types than RFs.
+        # 2. If we are reaching the maximum depth of repository folders
+        # (Configured in plone.registry), we should not be able to add
+        # any more RFs, but then we should be able to add the other configured
+        # types in any case. If the maximum_repository_depth is set to 0,
+        # we do not have a depth limit.
+
+        # Get the default types
+        types = super(Container, self.context).allowedContentTypes()
```
Comment (deiferni) [inline opengever/repository/constrains.py]: as discussed, please rewrite this 😉

File: opengever/repository/constrains.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        # Filter content types, if required
+        if not self.context.is_leaf_node():
+            # only allow same types
+            types = filter(lambda a: a == fti, types)
+
+        # Finally: remove not enabled resticted content types
+        marker_behavior = 'opengever.dossier.behaviors.restricteddossier.' + \
+            'IRestrictedDossier'
+
+        allowed = self.context.addable_dossier_types \
```
Comment (deiferni) [inline opengever/repository/constrains.py]: i think this is equivalent to

```python
allowed = self.context.addable_dossier_types
```

Comment (buchi) [inline opengever/repository/constrains.py]: It's not if `self.context.addable_dossier_types` is `None`. I prefer to keep the current implementation as-is and do refactoring or optimization in another PR.

**Comment by buchi** [comment]
@deiferni ready for rereview

File: opengever/repository/constrains.py (hunk 2 above)
Comment (deiferni) [inline opengever/repository/constrains.py]: ok fair enough 👍
